cases/detector_set.go

```diff
@@ -25,6 +28,21 @@ var (
 				&detector.ServiceNoAvailableTarget{},
 				&detector.ServicePartiallyAvailable{},
 			}
+
+			var err error
+			k8sVersionInfo, err := utils.GetK8sVersion(m[CONFIG_KUBECONFIG])
+			if err != nil {
+				return nil
+			}
+			k8sVersion, err := strconv.ParseFloat(k8sVersionInfo.Major+"."+k8sVersionInfo.Minor, 64)
+			if err != nil {
+				return nil
+			}
+			if k8sVersion >= K8S_VERSION_1_21 && k8sVersion < K8S_VERSION_1_25 {
+				detectors = append(detectors, &detector.ApiLifecyclePolicyV1Beta1{})
+			}
+
+			return detectors
```

Comment:
[Major]
This comment here is similar to the one in the Collector. However, in the case of the Detector, there are clearly situations where it should not be executed based on the Kubernetes version (to prevent false positive alerts).

What do you think about having each individual Detector check the [KeyK8sVersion](https://github.com/kakao/detek/blob/main/cases/collector/kubernetes_client.go#L79-L86) in their "Do" method and return a message indicating that everything is fine if the version condition doesn't match?

---below is the Korean version of the above comment---
[Major]
이쪽도 Collector 쪽과 같은 코멘트입니다. 다만 Detector 쪽은 Kubernetes 버전에 따라서 실행되지 않아야 하는 경우가 확실히 존재합니다. (아무런 이상이 없는데 버전이 달라서 문제가 있다는 경고가 날라갈 수 있기 때문에)

개별 Detector 쪽에서 `KeyK8sVersion` 를 보고 버전 조건이 맞지 않으면 이상 없다는 메세지를 반환하는 방법은 어떤가요?

Reply:
Okay. I changed the logic to do not consider Kubernetes version in detector set.

Instead, detect problems only if Kubernetes version is 1.21+ or higher in the indivisual case.
e.g)
In case of 1.19+, "policy/v1beta1 PodSecurityPolicy" is Normal.
In case of 1.23+, "policy/v1beta1 PodSecurityPolicy" is Warn.